webauthn/Sources/App/Controllers/WebAuthnController.swift

```diff
@@ -16,110 +16,126 @@ import FluentKit
 
 struct HBWebAuthnController {
+    typealias Context = WebAuthnRequestContext
+
     let webauthn: WebAuthnManager
+    let fluent: HBFluent
+    let sessionStorage: HBSessionStorage
 
-    func add(to group: HBRouterGroup) {
-        group
-            .post("signup", options: .editResponse, use: self.signin)
-            .get("login", options: .editResponse, use: self.beginAuthentication)
-        group
-            .add(middleware: WebAuthnSessionStateAuthenticator())
-            .post("register/start", use: self.beginRegistration)
-            .post("register/finish", use: self.finishRegistration)
-            .post("login", options: .editResponse, use: self.finishAuthentication)
-        group
-            .add(middleware: WebAuthnSessionAuthenticator())
-            .get("logout", options: .editResponse, use: self.logout)
+    // return RouteGroup with user login endpoints
+    var endpoints: some HBMiddlewareProtocol<Context> {
+        RouteGroup("user") {
+            Post("signup", handler: self.signin)
+            Get("login", handler: self.beginAuthentication)
+            Post("login") {
+                WebAuthnSessionStateAuthenticator(fluent: self.fluent, sessionStorage: self.sessionStorage)
+                self.finishAuthentication
+            }
+            Get("logout") {
+                WebAuthnSessionAuthenticator(fluent: self.fluent, sessionStorage: self.sessionStorage)
+                self.logout
+            }
+            RouteGroup("register") {
+                WebAuthnSessionStateAuthenticator(fluent: self.fluent, sessionStorage: self.sessionStorage)
+                Post("start", handler: self.beginRegistration)
+                Post("finish", handler: self.finishRegistration)
+            }
+        }
     }
 
     struct SignInInput: Decodable {
         let username: String
     }
 
-    func signin(request: HBRequest) async throws -> HBResponse {
-        let input = try request.decode(as: SignInInput.self)
-        guard try await User.query(on: request.db)
+    @Sendable func signin(request: HBRequest, context: Context) async throws -> HBResponse {
+        let input = try await request.decode(as: SignInInput.self, context: context)
+        guard try await User.query(on: self.fluent.db())
             .filter(\.$username == input.username)
             .first() == nil
         else {
             throw HBHTTPError(.conflict, message: "Username already taken.")
         }
         let user = User(username: input.username)
-        try await user.save(on: request.db)
-        let session = try WebAuthnSessionStateAuthenticator.Session.signedUp(userId: user.requireID())
-        try await request.session.save(
+        try await user.save(on: self.fluent.db())
+        let session = try WebAuthnSession.signedUp(userId: user.requireID())
+        let cookie = try await self.sessionStorage.save(
             session: session,
-            expiresIn: .minutes(10)
+            expiresIn: .seconds(600)
```

Comment:
Any reason for this change?

Reply:
`Duration` doesn't have a static function `minutes`